crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -135,12 +140,133 @@ impl Persistence {
+                .map_err(error::Auction::DbError)?
+                .ok_or(error::Auction::Missing)?
+                .into_iter()
+                .map(|order| domain::OrderUid(order.0))
+                .collect::<HashSet<_>>();
+
+            // get quotes for all auction orders
+            let quotes = self
+                .postgres
+                .read_quotes(auction_orders.iter())
+                .await
+                .map_err(error::Auction::DbError)?;
+
+            // get fee policies for all orders that were part of the competition auction
+            let policies = database::fee_policies::fetch_all_for_auction(&mut ex, auction_id)
+                .await
+                .context("fetch fee policies")
+                .map_err(error::Auction::DbError)?;
+
+            // compile order data
+            let mut orders = HashMap::new();
+            for order in auction_orders.iter() {
+                let quote = quotes
+                    .get(order)
+                    .ok_or(error::Auction::MissingQuote(*order))?;
+
+                orders.insert(*order, vec![]);
+                for policy in policies.get(&ByteArray(order.0)).unwrap_or(&vec![]) {
+                    let policy = dto::fee_policy::into_domain(policy.clone(), quote)
+                        .map_err(error::Auction::DbConversion)?;
+                    orders.entry(*order).and_modify(|policies| {
+                        policies.push(policy);
+                    });
+                }
+            }
+            orders
+        };
+
+        let surplus_capturing_jit_order_owners =
+            database::surplus_capturing_jit_order_owners::fetch(&mut ex, auction_id)
+                .await
+                .context("fetch surplus capturing jit order owners")
+                .map_err(error::Auction::DbError)?
+                .ok_or(error::Auction::MissingJitOrderOwners)?
+                .into_iter()
+                .map(|owner| eth::H160(owner.0).into())
+                .collect();
+
+        Ok(domain::settlement::Auction {
+            id: auction_id,
+            prices,
+            orders,
+            deadline,
+            surplus_capturing_jit_order_owners,
+        })
+    }
 }
 
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
     #[error("failed to read data from database")]
     DbError(#[from] anyhow::Error),
-    #[error(transparent)]
-    Price(#[from] domain::auction::InvalidPrice),
+}
+
+pub mod error {
+    use super::*;
+
+    #[derive(Debug, thiserror::Error)]
+    pub enum Auction {
+        #[error("failed to read data from database")]
+        DbError(anyhow::Error),
+        #[error("failed dto conversion from database")]
+        DbConversion(anyhow::Error),
+        #[error("auction data not found in the database")]
+        Missing,
+        #[error(transparent)]
+        Price(#[from] domain::auction::InvalidPrice),
+        #[error("quote not found in the database for an existing order")]
```

Comment:
Consider adding the order UID to the error log, something like:

```rust
#[error("quote not found in the database for an existing order: {0]")]
```

Reply:
added